### PR TITLE
Fix non terminal error

### PR DESCRIPTION
**Pull request by JPWerling**
### The What
A fix to the package not being able to run in non-terminal instances, such as cronjobs.

### The Why
Enables running the package non-terminal

### The How
Adds a check for connected terminal before trying to access attributes about it.

### This [update](https://semver.org/) is:
- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### Installation and initiation
* `git clone git@github.com:genomic-medicine-sweden/NGPIris.git`
* `cd NGPIris && git checkout <BRANCH>`
* `bash setup.sh`
* `source activate hpcenv`

### Tests
* `pytest tests/`
* Potential additional tests

### Expected outcome:
* Pytest resolves without crashes
* Potential additional results

## Confirmations:
- [ ] Code reviewed by @octocat
- [ ] Code tested by @octocat

### CODE REVIEW

File: NGPIris/hcp/hcp.py [hunk 1]
```diff
@@ -216,30 +219,14 @@ def search_objects(self, string):
         return [obj for obj in self.objects if string in obj.key]
 
     @bucketcheck
-    def upload_file(self, local_path, remote_key, metadata={},callback="default"):
+    def upload_file(self, local_path, remote_key, metadata={}):
```
Comment (sylvinite) [inline NGPIris/hcp/hcp.py]: Refering to our previous conversation.
Are you certain this works? Afaik pytests crash if `callback = ProgressPercentage(local_path)` is used.

**Review by sylvinite — CHANGES_REQUESTED** [review]
Most is agreeable. See single comment.

File: NGPIris/hcp/hcp.py (hunk 1 above)
Comment (JPWerling) [reply NGPIris/hcp/hcp.py]: Removed left-over references to the callback kwarg and all tests pass

**Comment by JPWerling** [comment]
The included changes will not be backwards compatible due to the removal of the callback kwarg. How would you like to proceed with the versioning? @sylvinite

**Comment by sylvinite** [comment]
We had a verbal dialogue and concluded the following:
- Callback should be still possible to be silenced, but better design is to use a boolean
- It would be good to have a test for callback silenced
- It would be good if the pytest generated files had random names to minimize testing conflicts

**Pull request merged**